### PR TITLE
aws.iam Document tidy and layout to match Principal, Action, Resource, Condition (PARC) order

### PR DESCRIPTION
**Pull request by 0xjjoyy**
*Issue #, if available:*

*Description of changes:*

1. Layout to match Principal, Action, Resource, Condition (PARC) order
2. Tidy the condition key traits layout
3. Update to use `myservice` instead of `otherservice` for condition keys, as condition keys should reference the same service.
4. Capitalization of titles

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/source-2.0/aws/aws-iam.rst [hunk 1]
```diff
@@ -369,13 +312,79 @@ operation for it to complete successfully.
...
+
 
 .. _aws-iam_traits-condition-keys:
 
 --------------------
-Condition key traits
+Condition Key Traits
 --------------------
 
+
```
Comment (hpmellema) [inline docs/source-2.0/aws/aws-iam.rst:384]: nit: extra whitespace?

**Review by hpmellema — APPROVED** [review]

File: docs/source-2.0/aws/aws-iam.rst [hunk 2]
```diff
@@ -18,12 +18,17 @@ Smithy are automatically inferred. These can be disabled or augmented. For
 more information, see :ref:`deriving-condition-keys`.
 
 
+.. _aws-iam_traits-principal:
+
+----------------
+Principal Traits
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:24]:
```suggestion
Principal traits
```

Our spec uses titles like this, [see "Constraint traits"](https://smithy.io/2.0/spec/constraint-traits.html).

File: docs/source-2.0/aws/aws-iam.rst [hunk 3]
```diff
@@ -64,72 +69,10 @@ The following example defines two operations:
...
-            superId: String,
-        }
-    }
-
-
 .. _aws-iam_traits-actions:
 
 -------------
-Action traits
+Action Traits
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:75]:
```suggestion
Action traits
```

File: docs/source-2.0/aws/aws-iam.rst [hunk 4]
```diff
@@ -369,13 +312,79 @@ operation for it to complete successfully.
...
+        }
+    }
+
+
 
 .. _aws-iam_traits-condition-keys:
 
 --------------------
-Condition key traits
+Condition Key Traits
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:381]:
```suggestion
Condition key traits
```

File: docs/source-2.0/aws/aws-iam.rst [hunk 5]
```diff
@@ -369,13 +312,79 @@ operation for it to complete successfully.
...
 --------------------
 
+
+
+
+
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:387]:
```suggestion
```

File: docs/source-2.0/aws/aws-iam.rst [hunk 6]
```diff
@@ -715,6 +602,49 @@ explicitly binds ``ActionContextKey1`` to the field ``key``.
...
+    * - ``ArrayOfIPAddress``
+      - An unordered list of IPAddress types.
+    * - ``ArrayOfNumeric``
+      - An unordered list of Numeric types.
+    * - ``ArrayOfString``
+      - An unordered list of String types.
+
 .. _deriving-condition-keys:
 
 Deriving Condition Keys
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:650]:
```suggestion
Deriving condition keys
```

File: docs/source-2.0/aws/aws-iam.rst [hunk 7]
```diff
@@ -808,6 +738,87 @@ The computed condition keys for the service are:
           * ``aws:region``
 
 
+
+.. smithy-trait:: aws.iam#disableConditionKeyInference
+.. _aws.iam#disableConditionKeyInference-trait:
+
+``aws.iam#disableConditionKeyInference`` trait
```
Comment (kstich) [inline docs/source-2.0/aws/aws-iam.rst:745]: This trait needs to be before the "Deriving condition keys" section, as it is mentioned there.

**Review by kstich — CHANGES_REQUESTED** [review]

File: docs/source-2.0/aws/aws-iam.rst (hunk 1 above)
Comment (0xjjoyy) [reply docs/source-2.0/aws/aws-iam.rst]: accepted the proposed change below

**Pull request closed without merging**